openvino_xai/methods/white_box/torch.py

```diff
@@ -153,16 +205,17 @@ def _feature_hook(self, module: torch.nn.Module, inputs: Any, output: torch.Tens
         return torch.cat(feature_maps)
 
     def _output_hook(self, module: torch.nn.Module, inputs: Any, output: torch.Tensor) -> Dict[str, torch.Tensor]:
-        batch_size, _, h, w = self._feature_shape
-        num_classes = output.shape[1]
-        predictions = output[:batch_size]
-        saliency_maps = output[batch_size:]
-        saliency_maps = saliency_maps.reshape([batch_size, h * w, num_classes])
-        saliency_maps = saliency_maps.transpose(1, 2)  # BxHWxC -> BxCxHW
+        """Split combined output B0xC into BxC precition and BxCxHxW saliency map."""
+        batch_size, _, h, w = self._feature_shape  # B0xDxHxW
```

Comment:
```suggestion
        batch_size, _, h, w = self._feature_shape  # B0x_xHxW
```

Reply:
Thank you for adding more clarity here!

Reply:
I'll keep it as D as it's common to denote the feature dimension.